certora/specs/Health.spec

```
@@ -0,0 +1,115 @@
+
+    function _.price() external => constantPrice expect uint256;
+
+    function marketParams() internal returns (PreLiquidation.MarketParams memory)
+        => summaryMarketParams();
+
+    function MORPHO.extSloads(bytes32[]) external returns bytes32[] => NONDET DELETE;
+    function MORPHO.market(PreLiquidation.Id) external
+      returns (uint128, uint128, uint128,uint128, uint128, uint128) envfree;
+    function MORPHO.position(PreLiquidation.Id, address) external
+      returns (uint256, uint128, uint128) envfree;
+    function MORPHO.totalBorrowShares(PreLiquidation.Id) external returns (uint256) envfree;
+    function MORPHO.totalBorrowAssets(PreLiquidation.Id) external returns (uint256) envfree;
+
+    function UtilsLib.exactlyOneZero(uint256 a, uint256 b) internal
+        returns bool => summaryExactlyOneZero(a,b);
+    function Util.oraclePriceScale() external returns (uint256) envfree;
+    function Util.wad() external returns (uint256) envfree;
+
+    function MORPHO.accrueInterest(PreLiquidation.MarketParams) external => NONDET;
+    function Morpho._isHealthy(MorphoHarness.MarketParams memory, MorphoHarness.Id,address) internal returns (bool) => NONDET;
+    function Morpho._accrueInterest(MorphoHarness.MarketParams memory, MorphoHarness.Id) internal => NONDET;
+
+    function preLIF(address) external returns uint256 envfree;
+    function preLCF(address) external returns uint256 envfree;
+}
+
+rule preLIFBounded(env e, address borrower){
+    // Avoid division by zero.
+    requireInvariant preLltvConsistent();
+    requireInvariant preLCFConsistent();
+    requireInvariant preLIFConsistent();
+    requireInvariant hashOfMarketParamsOf();
+
+    // We place ourselves at the last block for getting the following variables.
+    require MORPHO.lastUpdate(currentContract.ID) == e.block.timestamp;
+    require borrower != 0;
+
+    assert currentContract.PRE_LIF_1 <= preLIF(borrower);
+    assert preLIF(borrower) <= currentContract.PRE_LIF_2;
+}
+
+rule preLCFBounded(env e, address borrower){
+    requireInvariant preLltvConsistent();
+    requireInvariant preLCFConsistent();
+    requireInvariant preLIFConsistent();
+    requireInvariant hashOfMarketParamsOf();
+    require MORPHO.lastUpdate(currentContract.ID) == e.block.timestamp;
+    require borrower != 0;
+
+    assert currentContract.PRE_LCF_1 <= preLCF(borrower);
+    assert preLCF(borrower) <= currentContract.PRE_LCF_2;
+}
+
+rule positionDoesntDegrade(env e,address borrower, uint256 seizedAssets, bytes data) {
+    // Avoid division by zero.
+    requireInvariant preLltvConsistent();
+    requireInvariant preLCFConsistent();
+    requireInvariant preLIFConsistent();
+    requireInvariant hashOfMarketParamsOf();
+
+    // We place ourselves at the last block for getting the following variables.
+    require MORPHO.lastUpdate(currentContract.ID) == e.block.timestamp;
+
+    // Assume no callback.
+    require data.length == 0;
```

```suggestion
    // Assume no callback.
    require data.length == 0;

    // We place ourselves at the last block for getting the following variables.
    require MORPHO.lastUpdate(currentContract.ID) == e.block.timestamp;
```
So that the comment "We place ourselves at the last block for getting the following variables." is at the right place